e2e/libs/node/node.py

```
@@ -120,6 +120,18 @@ def get_node_cpu_cores(self, node_name):
 
+    def get_node_total_memory(self, node_name):
+        node = self.get_node_by_name(node_name)
+        return node.status.capacity['memory']
+
+    def get_node_condition(self, node_name, condition_type):
+        node = self.get_node_by_name(node_name)
+        for condition in node.status.conditions:
+            if condition.type == condition_type:
+                logging(f"Got node {node_name} condition {condition_type}: {condition}")
+                return condition.status
+        assert False, f"Failed to get node {node_name} condition {condition_type}: {node}"
```

Ruff (0.8.2)

Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)